src/spirc.rs

```diff
@@ -170,6 +172,16 @@ impl<D: SpircDelegate> SpircManager<D> {
             protocol::spirc::MessageType::kMessageTypePause => {
                 self.delegate.pause();
             }
+            protocol::spirc::MessageType::kMessageTypeNext => {
+            	self.index = (self.index + 1) % self.tracks.len() as u32;
+                let track = self.tracks[self.index as usize];
```

Comment:
Wow thanks for the extensive note.  Having a Quick look i cant see any way of determining the actual queue order. 
The queue is not implemented at the moment as far as i can see. I could give it a shot after fixing the pull request. In the mean time, could you tell me how you got that protobuf output? Thanks in advance